public/utils/message_content_pool.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { BehaviorSubject, Observable, Subscriber } from 'rxjs';
+
+const JOB_INTERVAL = 50;
+const CONTENT_SLICE_LENGTH = 10;
+
+export class MessageContentPool {
+  // Backend may give large response in a single chunk,
+  // use an observable here to buffer the large response.
+  private messageContentChunk$: BehaviorSubject<Record<string, string>>;
+  private messageContentTimer: NodeJS.Timeout | null = null;
+  private outputSubscriber: Subscriber<{ messageId: string; messageContent: string }> | null = null;
+  private output$: Observable<{ messageId: string; messageContent: string }>;
+  private inputCompleted: boolean = false;
+  constructor() {
+    this.messageContentChunk$ = new BehaviorSubject<Record<string, string>>({});
+    this.output$ = new Observable((subscriber) => {
+      this.outputSubscriber = subscriber;
+    });
+  }
+
+  private startMessageContentJob() {
+    this.messageContentTimer = setTimeout(() => {
+      const restContents = Object.entries(this.messageContentChunk$.getValue()).reduce(
+        (acc, cur) => {
+          const [messageId, messageContent] = cur;
+          this.outputSubscriber?.next({
+            messageId,
+            messageContent: messageContent.slice(0, CONTENT_SLICE_LENGTH),
+          });
+          const restContent = messageContent.slice(CONTENT_SLICE_LENGTH);
+          if (restContent.length > 0) {
+            return {
+              ...acc,
+              [messageId]: restContent,
+            };
+          }
+          return acc;
```

Comment:
Do we need to clear current content if `resetContent.length` equal 0? 
For example, the value of `messageContentChunk$`
```
{"message-1":"foo"}
```
The `outputSubscriber` will emit '{messageId:"message-1", messageContent: "foo"}'.
Since the restContent.length equal zero, it will return original `acc`. The messageContent for "message-1" still "foo".